### PR TITLE
Support open compound nouns

### LINKED ISSUE #108: list index out of range in engine.postprocess
Inflect pip installed into virtual environment running python3.7 on ubuntu 18.04

After getting an error I wrote some debug statements into inflect.py starting on line 2215:

        def postprocess(self, orig, inflected):
            inflected = str(inflected)

            print("Inflected: " + inflected)
            if "|" in inflected:
                inflected = inflected.split("|")[self.classical_dict["all"]]
            result = inflected.split(" ")

            print("Orig: " + orig)
            print("Inflected: " + inflected)

            # Try to fix word wise capitalization
            for index, word in enumerate(orig.split(" ")):
                ...

My code:

    >>> import inflect
    >>> p = inflect.engine()
    >>> p.classical(all=True)
    >>> p.plural("Genie")
    Inflected: genies|genii
    Orig: Genie
    Inflected: genii
    'Genii'

    >>> p.plural("Master Genie")
    Inflected: Master genies|genii
    Orig: Master Genie
    Inflected: genii

    Traceback (most recent call last):
      File "<input>", line 1, in <module>
      File "/omitted/for/privacy/lib/python3.7/site-packages/inflect.py", line 2260, in plural
        or self._plnoun(word, count),
      File "/omitted/for/privacy/lib/python3.7/site-packages/inflect.py", line 2230, in postprocess
        result[index] = result[index].capitalize()
    IndexError: list index out of range

So it looks like something went wrong in this line:
    inflected = inflected.split("|")[self.classical_dict["all"]]

Instead of returning "master genii" it returns just "genii"

**Ninja Edit**: This would be occuring on line 2226 in postprocess without the debug statements

### PR DESCRIPTION
**Pull request by theauk**
`postprocess()` has been updated to support open/spaced compound nouns where the second word has a classical plural. The inner if-statement is required to check if both parts (e.g. "prima donna") or only the last part (e.g. "master genie") need to be pluralized. 

Closes #108

### CODE REVIEW

File: inflect.py [hunk 1]
```diff
@@ -2286,8 +2286,19 @@ def inflect(self, text: str) -> str:
...
+            word_options = inflected.split("|")
+            # When two parts of a noun need to be pluralized
+            if len(word_options[0].split(" ")) == len(word_options[1].split(" ")):
+                result = inflected.split("|")[self.classical_dict["all"]].split(" ")
+            # When only the last part of the noun needs to be pluralized
+            else:
+                result = inflected.split(" ")
+                for index, word in enumerate(result):
+                    if "|" in word:
+                        result[index] = word.split("|")[self.classical_dict["all"]]
```
Comment (jaraco) [inline inflect.py:2298]: I was originally thinking that this approach could be replaced with a comprehension, something like this:

```suggestion
                result = [
                    word.split("|")[self.classical_dict["all"]] if "|" in word else word
                    for word in inflected.split(" ")
                ]
```

But I'm not sure that reads any better. What do you think?

**Review by jaraco — APPROVED** [review]
Looks great. Without the tests, I would be totally unsure about this change, but with them, it's great. Let me know what you think about the comprehension.

I'm going to add a changelog entry the PR so it's ready for release after merge.

File: inflect.py (hunk 1 above)
Comment (theauk) [reply inflect.py:2298]: It's definitely more concise. Hadn't seen exactly that type of comprehension before, so had to read it a couple of times – but I think it works fine here.

Comment (jaraco) [reply inflect.py:2298]: In that case, let's just go with the original approach.

**Pull request merged**